lib/hocon/impl/config_impl.rb

```diff
@@ -191,15 +191,44 @@ def self.trace(message, indent_level = 0)
     $stderr.puts(message)
   end
 
+  def self.empty_object_from_origin(origin)
+    # we want null origin to go to SimpleConfigObject.empty() to get the
+    # origin "empty config" rather than "hardcoded value"
+    if origin == @default_value_origin
+      @default_empty_object
+    else
+      Hocon::Impl::SimpleConfigObject.empty(origin)
+    end
+  end
+
+  def empty_object(origin_description)
+    if !origin_description.nil?
+      origin = Hocon::Impl::SimpleConfigOrigin.new_simple(origin_description)
+    else
+      origin = nil
+    end
+
+    empty_object_from_origin(origin)
+  end
+
+  def self.empty_config(origin_description)
+    empty_object(origin_description).to_config
+  end
+
+  def empty(origin_description)
+    empty_object(origin_description).to_config
```

Comment:
I think the parameter here should be `origin`, and that this should call `empty_object_from_origin`.